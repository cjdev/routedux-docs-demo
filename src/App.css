.App {
    text-align: center;
}

.App-logo {
    animation: App-logo-spin infinite 20s linear;
    height: 80px;
}

.App-header {
    background-color: #222;
    height: 150px;
    padding: 20px;
    color: white;
}

.App-intro {
    font-size: large;
}

@keyframes App-logo-spin {
    from {
        transform: rotate(0deg);
    }
    to {
        transform: rotate(360deg);
    }
}

.article-body ul {
    padding-left: 1.5em;
}

.article-body h3 {
    font-weight: bold;
}

.article-body {
    max-width: 80em;
}

.article-body h1 {
    padding-bottom: 0.5em;
}
