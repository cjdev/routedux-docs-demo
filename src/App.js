import React from "react";
import "./App.css";
import "@cjdev/visual-stack/lib/global";
import { VIEW_ARTICLE } from "./model/actions";
import { SideNav } from "@cjdev/visual-stack/lib/components/SideNav";
import { ActionLink } from "routedux";
import { Panel, Body, Header } from "@cjdev/visual-stack/lib/components/Panel";
import ApplicationLayout from "@cjdev/visual-stack/lib/layouts/ApplicationLayout";
import { connect } from "react-redux";
import { getContentsById } from "./model";

function App({ title, contents, tableOfContents }) {
  const menu = (
    <SideNav onClick={() => {}}>
      {tableOfContents.map(item => {
        return (
          <ActionLink
            key={item.id}
            action={{ type: VIEW_ARTICLE, id: item.id }}
          >
            {item.title}
          </ActionLink>
        );
      })}
    </SideNav>
  );

  return (
    <ApplicationLayout sideNav={menu}>
      <Panel>
        <Header>{title}</Header>
        <div className="article-body">
          <Body>
            <div dangerouslySetInnerHTML={{ __html: contents }} />
          </Body>
        </div>
      </Panel>
    </ApplicationLayout>
  );
}

const mapStateToProps = (
  { currentTitle: title, id, ...state },
  { tableOfContents }
) => {
  const foundContents = getContentsById(id, state);
  const contents = foundContents ? foundContents : "Loading...";

  return { title, contents, tableOfContents };
};

export default connect(mapStateToProps)(App);

export { App, mapStateToProps };
