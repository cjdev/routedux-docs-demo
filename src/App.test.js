"use strict";
import 'raf/polyfill';
import React from 'react';
import {shallow} from 'enzyme';
import ezJson from 'enzyme-to-json';
import {App, mapStateToProps} from './App';

import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({ adapter: new Adapter() });

const tableOfContents = [{"contentsUrl":"/pages/bar.html","id":"bar","title":"A Bar's Life"},{"contentsUrl":"/pages/foo.html","id":"foo","title":"Foo"},{"contentsUrl":"/pages/home.html","id":"home","title":"Our Doc Site"}];

it('renders side nav with correct links, and a header and content passed in', () => {
  // given

  // when
  const wrapper = shallow(
    <App tableOfContents={tableOfContents} title="Hello World" contents="<div>Some html</div>" />
  );

  // then
  expect(ezJson(wrapper)).toMatchSnapshot();

});

it("mapStateToProps returns expected props from store", () => {

  // given
  const id = "woo";
  const title = "A great title";
  const state = {
    currentTitle: title,
    id: id,
    contentMap: {
      "woo": "Woo's Contents",
    }
  };
  const dflt = "Loading...";

  // when
  const {
    title: actualTitle,
    contents: actualContents,
    tableOfContents: actualTableOfContents,
  } = mapStateToProps({currentTitle: title, id, ...state}, {tableOfContents});

  const {
    contents: actualFailContents,
  } = mapStateToProps({...state, currentTitle: title, id: id+"-missing"}, {tableOfContents});

  // then
  expect(actualTitle).toEqual(title);
  expect(actualContents).toEqual(state.contentMap[id]);
  expect(actualFailContents).not.toEqual(state.contentMap[id]);
  expect(actualTableOfContents).toEqual(tableOfContents);
});