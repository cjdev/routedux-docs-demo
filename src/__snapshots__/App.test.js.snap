// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders side nav with correct links, and a header and content passed in 1`] = `
<ApplicationLayout
  sideNav={
    <SideNav>
      <ActionLink
        action={
          Object {
            "id": "bar",
            "type": "VIEW_ARTICLE",
          }
        }
      >
        A Bar's Life
      </ActionLink>
      <ActionLink
        action={
          Object {
            "id": "foo",
            "type": "VIEW_ARTICLE",
          }
        }
      >
        Foo
      </ActionLink>
      <ActionLink
        action={
          Object {
            "id": "home",
            "type": "VIEW_ARTICLE",
          }
        }
      >
        Our Doc Site
      </ActionLink>
    </SideNav>
  }
>
  <Panel>
    <Header>
      Hello World
    </Header>
    <div
      className="article-body"
    >
      <Body>
        <div
          dangerouslySetInnerHTML={
            Object {
              "__html": "<div>Some html</div>",
            }
          }
        />
      </Body>
    </div>
  </Panel>
</ApplicationLayout>
`;
